lib/js/components/ProgressBar/ProgressBar.vue

```diff
@@ -0,0 +1,330 @@
+
+				&.-secondary {
+					background: map-get($color-map, 'ghost');
+				}
+			}
+
+			&.-schemeMedium {
+				#{$self}__result {
+					background: map-get($color-map, 'medium');
+
+					&.-secondary {
+						background: map-get($color-map, 'weak');
+					}
+				}
+			}
+
+			&.-schemeMediumNeutral {
+				#{$self}__result {
+					background: map-get($color-map, 'medium');
+
+					&.-secondary {
+						background: map-get($color-map, 'neutral-weak');
+					}
+				}
+			}
+		}
+	}
+
+	&-noRadius {
+		border-radius: 0;
+	}
+
+	&.-compact {
+		#{$self}__labelText {
+			@include headlineXS();
+		}
+		#{$self}__label {
+			margin-bottom: $space-xxxs;
+		}
+	}
+
+	&__bar {
+		background-color: $color-default-background;
+		border-radius: $progress-bar-border-radius;
+		box-shadow: inset 0 1px 4px $color-default-shadow-heavy;
+		height: $progress-bar-height;
+		overflow: hidden;
+
+		&.-small {
+			height: $progress-bar-s-height;
+		}
+
+		&.-xsmall {
+			border-radius: 0;
+			box-shadow: inset 0 1px 3px $color-default-shadow-heavy;
+			height: $progress-bar-xs-height;
+		}
+
+		&.-noRadius {
+			border-radius: 0;
+		}
+	}
+
+	&__label {
+		align-items: flex-end;
+		display: flex;
+		justify-content: space-between;
+		margin-bottom: $space-xxxs;
+
+		@media #{breakpoint-s()} {
+			margin-bottom: $space-xxs;
+		}
+	}
+
+	&__labelText {
+		@include headlineXS();
+		@include textBold;
+
+		color: $color-neutral-text-heavy;
+
+		@media #{breakpoint-s()} {
+			@include headlineS();
+		}
+	}
+
+	&__labelDataWrapper {
+		@include textS();
+		@include textBold;
+
+		align-items: baseline;
+		color: $color-neutral-text-heavy;
+		display: flex;
+		margin-left: $space-xxs;
+	}
+
+	&__labelDataSupporting {
+		color: $color-neutral-text;
+		display: flex;
+		margin-left: $space-xxxxxs;
+	}
+
+	&__labelDataSeparator {
+		margin-right: $space-xxxxxs;
+	}
+
+	&__labelDataSuffix {
+		@include textXS;
+
+		color: $color-neutral-text-weak;
+		margin-left: $space-xxxxs;
+	}
+
+	&__results {
+		height: 100%;
+		position: relative;
+		width: 100%;
+	}
+
+	&__result {
+		height: 100%;
+		position: absolute;
+		top: 0;
+	}
+}
+</style>
+
+<script lang="ts">
+import {
+	PROGRESS_BAR_COLORS,
+	PROGRESS_BAR_SIZES,
+	PROGRESS_BAR_RADII,
+	PROGRESS_BAR_LAYOUTS,
+	PROGRESS_BAR_COLOR_SCHEMES,
+} from './ProgressBar.consts';
+
+export default {
+	name: 'ProgressBar',
+	props: {
+		numberOfLayers: {
+			type: Number,
+			default: 1,
+		},
+		colorScheme: {
+			type: String,
+			default: PROGRESS_BAR_COLOR_SCHEMES.DEFAULT,
+			validate(size) {
+				return Object.values(PROGRESS_BAR_COLOR_SCHEMES).includes(size);
+			},
+		},
+		color: {
+			type: String,
+			required: true,
+			validate(color) {
+				return Object.values(PROGRESS_BAR_COLORS).includes(color);
+			},
+		},
+		size: {
+			type: String,
+			default: PROGRESS_BAR_SIZES.MEDIUM,
+			validate(size) {
+				return Object.values(PROGRESS_BAR_SIZES).includes(size);
+			},
+		},
+		ranges: {
+			type: Array,
```

Comment:
Do we want to defined PropType for range to make it easier to guess what are required attributes here?

Reply:
Ok, I will do that.